### PR TITLE
[10.0] Add google_city_validation

### PR DESCRIPTION
**Pull request by yasmineelmrini**

### CODE REVIEW

File: google_city_validation/data/ir_config_parameter.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+  <data>
```
Comment (joaoalf) [inline google_city_validation/data/ir_config_parameter.xml]: @rimbendhaou It's better to use `noupdate` here.

File: google_city_validation/models/res_config.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    city_validation_policy = fields.Selection(
+        [('soft', 'Soft'), ('strict', 'Strict')],
+        help='Soft only warn the user / Strict delete the value of the city',
+    )
+
+    @api.model
+    def get_default_policy(self, fields):
+        policy = self.env['ir.config_parameter'].sudo(
+        ).get_param('google_city_validation.policy')
```
Comment (joaoalf) [inline google_city_validation/models/res_config.py]: @rimbendhaou indentation?

File: google_city_validation/static/src/js/google_city_validation.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            this._super();
+            this.geocoder = new google.maps.Geocoder();
+        },
+        onchangeCity: function() {
+            var param = new Model('ir.config_parameter');
+            var request = {'address': this.get_value()};
+            var self = this;
+            param.call('get_param', ['google_city_validation.policy', false])
+            .then(function(policy) {
+                console.log('policy')
```
Comment (joaoalf) [inline google_city_validation/static/src/js/google_city_validation.js]: @rimbendhaou ??

File: google_city_validation/static/src/js/google_city_validation.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+                console.log(policy)
```
Comment (joaoalf) [inline google_city_validation/static/src/js/google_city_validation.js]: @rimbendhaou ??

File: google_city_validation/static/src/js/google_city_validation.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+                if (self.get_value() != '') {
+                    self.geocoder.geocode(request, function(results, status) {
+                        if (status != 'OK' ||
+                         !results[0].types.includes("locality")) {
+                             if (policy === 'strict') {
+                                 console.log('strict')
```
Comment (joaoalf) [inline google_city_validation/static/src/js/google_city_validation.js]: @rimbendhaou ??

File: google_city_validation/static/src/js/google_city_validation.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+                                 self.set_value('')
+                                 alert("This city doesn't exist.");
```
Comment (joaoalf) [inline google_city_validation/static/src/js/google_city_validation.js]: @rimbendhaou I'm not sure that `alert()` is the proper odoo way to return a msg to the user.

File: google_city_validation/static/src/js/google_city_validation.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+                             }
+                             else {
+                                 console.log('soft')
```
Comment (joaoalf) [inline google_city_validation/static/src/js/google_city_validation.js]: @rimbendhaou ??

**Review by joaoalf — CHANGES_REQUESTED** [review]

**Comment by joaoalf** [comment]
@rimbendhaou Can you just clean-up a bit the git history? We should not see the merge commit and you add an internal ticket number in the commit prefix.

**Comment by rimbendhaou** [comment]
@joaoalf ok

**Review by joaoalf — APPROVED** [review]

**Pull request merged**